### ISSUE TITLE
Can I change the activation layer in the YOLOv8 Model?

### ISSUE THREAD

**Issue by FONG1234** [ultralytics/ultralytics#2417]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

In my project, I want to test out the performance of YOLOv8 with different activation function on my custom dataset, but I am not sure how to change the activation function and what the activation function YOLOv8 provided. Thanks for the people who willing to answer my question.

**Comment by glenn-jocher** [comment]
@FONG1234 hi! Thanks for your interest in YOLOv8. By default, YOLOv8 uses Mish activation function which was shown in recent study to improve CNN performance. YOLOv8 repo contains a file named 'models/common.py' that lists all the available activations, including LeakyReLU, ReLU, SiLU and more. You can use any of those activations instead of the default Mish activation by simply changing the activation parameter to your selected activation function name in the model configuration file. If you have any further questions or concerns, please don't hesitate to ask.

**Comment by FONG1234** [comment]
Oh I see, nice! Can I know how to install the repo for the YOLOv8, because when follow the instruction on training using the YOLOv8 model, there is no repo generated for YOLOv8. Thank you.

**Comment by FONG1234** [comment]
Sorry, I still can't find the location to change the activation function, because I cannot find the file of models/common.py. Do there any website I can refer to? Appreciated.

**Comment by Laughing-q** [comment]
@FONG1234 actually I think you can just add act layer in your model yaml file, here's a relu example:
```yaml
# Ultralytics YOLO 🚀, AGPL-3.0 license
# YOLOv8 object detection model with P3-P5 outputs. For Usage examples see https://docs.ultralytics.com/tasks/detect

# Parameters
nc: 80  # number of classes
act: nn.ReLU()
scales: # model compound scaling constants, i.e. 'model=yolov8n.yaml' will call yolov8.yaml with scale 'n'
  # [depth, width, max_channels]
  n: [0.33, 0.25, 1024]  # YOLOv8n summary: 225 layers,  3157200 parameters,  3157184 gradients,   8.9 GFLOPs
  s: [0.33, 0.50, 1024]  # YOLOv8s summary: 225 layers, 11166560 parameters, 11166544 gradients,  28.8 GFLOPs
  m: [0.67, 0.75, 768]   # YOLOv8m summary: 295 layers, 25902640 parameters, 25902624 gradients,  79.3 GFLOPs
  l: [1.00, 1.00, 512]   # YOLOv8l summary: 365 layers, 43691520 parameters, 43691504 gradients, 165.7 GFLOPs
  x: [1.00, 1.25, 512]   # YOLOv8x summary: 365 layers, 68229648 parameters, 68229632 gradients, 258.5 GFLOPs

# YOLOv8.0n backbone
backbone:
  # [from, repeats, module, args]
  - [-1, 1, Conv, [64, 3, 2]]  # 0-P1/2
  - [-1, 1, Conv, [128, 3, 2]]  # 1-P2/4
  - [-1, 3, C2f, [128, True]]
  - [-1, 1, Conv, [256, 3, 2]]  # 3-P3/8
  - [-1, 6, C2f, [256, True]]
  - [-1, 1, Conv, [512, 3, 2]]  # 5-P4/16
  - [-1, 6, C2f, [512, True]]
  - [-1, 1, Conv, [1024, 3, 2]]  # 7-P5/32
  - [-1, 3, C2f, [1024, True]]
  - [-1, 1, SPPF, [1024, 5]]  # 9

# YOLOv8.0n head
head:
  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 6], 1, Concat, [1]]  # cat backbone P4
  - [-1, 3, C2f, [512]]  # 12

  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 4], 1, Concat, [1]]  # cat backbone P3
  - [-1, 3, C2f, [256]]  # 15 (P3/8-small)

  - [-1, 1, Conv, [256, 3, 2]]
  - [[-1, 12], 1, Concat, [1]]  # cat head P4
  - [-1, 3, C2f, [512]]  # 18 (P4/16-medium)

  - [-1, 1, Conv, [512, 3, 2]]
  - [[-1, 9], 1, Concat, [1]]  # cat head P5
  - [-1, 3, C2f, [1024]]  # 21 (P5/32-large)

  - [[15, 18, 21], 1, Detect, [nc]]  # Detect(P3, P4, P5)
```

**Comment by FONG1234** [comment]
@Laughing-q Oh I see, is it like this?

![Activation Layer](https://github.com/ultralytics/ultralytics/assets/132126352/53b09e1b-3836-42dd-821a-3747ced36bfe)

By the way can I know, by adding this code, the entire activation layer will change to ReLU layer or just particular layers are changed to ReLU layer? Besides, actually how much different types activation function that I can change or there is any website I can refer?

Thank You So Much =)

**Comment by Laughing-q** [comment]
@FONG1234 entire activation layer

**Comment by FONG1234** [comment]
OKOK got it, Thank You So Much!!

**Comment by glenn-jocher** [comment]
@FONG1234 you're welcome! To answer your question, you can change the activation function in the model configuration file by adding the appropriate activation layer to the YAML file and passing the name of the activation function as a parameter. By doing so, it will change the activation layer of the entire model architecture to the designated activation function. Regarding the types of activation functions that you can change, you can refer to the available activation functions in PyTorch documentation, where you can find various activation functions such as ReLU, LeakyReLU, Sigmoid, and more.

**Comment by mnicnc404** [comment]
> @FONG1234 actually I think you can just add act layer in your model yaml file, here's a relu example:
> 
> ```yaml
> # Ultralytics YOLO 🚀, AGPL-3.0 license
> # YOLOv8 object detection model with P3-P5 outputs. For Usage examples see https://docs.ultralytics.com/tasks/detect
> 
> # Parameters
> nc: 80  # number of classes
> act: nn.ReLU()
> scales: # model compound scaling constants, i.e. 'model=yolov8n.yaml' will call yolov8.yaml with scale 'n'
>   # [depth, width, max_channels]
>   n: [0.33, 0.25, 1024]  # YOLOv8n summary: 225 layers,  3157200 parameters,  3157184 gradients,   8.9 GFLOPs
>   s: [0.33, 0.50, 1024]  # YOLOv8s summary: 225 layers, 11166560 parameters, 11166544 gradients,  28.8 GFLOPs
>   m: [0.67, 0.75, 768]   # YOLOv8m summary: 295 layers, 25902640 parameters, 25902624 gradients,  79.3 GFLOPs
>   l: [1.00, 1.00, 512]   # YOLOv8l summary: 365 layers, 43691520 parameters, 43691504 gradients, 165.7 GFLOPs
>   x: [1.00, 1.25, 512]   # YOLOv8x summary: 365 layers, 68229648 parameters, 68229632 gradients, 258.5 GFLOPs
> 
> # YOLOv8.0n backbone
> backbone:
>   # [from, repeats, module, args]
>   - [-1, 1, Conv, [64, 3, 2]]  # 0-P1/2
>   - [-1, 1, Conv, [128, 3, 2]]  # 1-P2/4
>   - [-1, 3, C2f, [128, True]]
>   - [-1, 1, Conv, [256, 3, 2]]  # 3-P3/8
>   - [-1, 6, C2f, [256, True]]
>   - [-1, 1, Conv, [512, 3, 2]]  # 5-P4/16
>   - [-1, 6, C2f, [512, True]]
>   - [-1, 1, Conv, [1024, 3, 2]]  # 7-P5/32
>   - [-1, 3, C2f, [1024, True]]
>   - [-1, 1, SPPF, [1024, 5]]  # 9
> 
> # YOLOv8.0n head
> head:
>   - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
>   - [[-1, 6], 1, Concat, [1]]  # cat backbone P4
>   - [-1, 3, C2f, [512]]  # 12
> 
>   - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
>   - [[-1, 4], 1, Concat, [1]]  # cat backbone P3
>   - [-1, 3, C2f, [256]]  # 15 (P3/8-small)
> 
>   - [-1, 1, Conv, [256, 3, 2]]
>   - [[-1, 12], 1, Concat, [1]]  # cat head P4
>   - [-1, 3, C2f, [512]]  # 18 (P4/16-medium)
> 
>   - [-1, 1, Conv, [512, 3, 2]]
>   - [[-1, 9], 1, Concat, [1]]  # cat head P5
>   - [-1, 3, C2f, [1024]]  # 21 (P5/32-large)
> 
>   - [[15, 18, 21], 1, Detect, [nc]]  # Detect(P3, P4, P5)
> ```

I'm using `8.0.124` and I think this is outdated. After some experiments, the correct keyword should be `activation`. For example:
```yaml
# Parameters
nc: 80  # number of classes
activation: "nn.LeakyReLU(0.02, inplace=True)"
...
```
Hi @glenn-jocher , is this written document? If so, where is it documented? I check [ultralytics official documentation website](https://docs.ultralytics.com/) and cannot find it. Actually, I found it from your codebase in [ultralytics/models/v6/yolov6.yaml](https://github.com/ultralytics/ultralytics/blob/cd0bf05aa476c3499d72fab1407d011218124a5f/ultralytics/models/v6/yolov6.yaml#L6).

**Comment by glenn-jocher** [comment]
Hi @mnicnc404, thanks for your question and for bringing up the confusion about the correct keyword for changing the activation function in the YOLOv8 model configuration. I apologize for the outdated code example in the previous response.

To change the activation function in the YOLOv8 model, you should use the keyword `activation` instead of `act` in the model YAML file. For example, you can specify the activation function as `'nn.LeakyReLU(0.02, inplace=True)'`.

I understand that you couldn't find this documented in the official Ultralytics documentation. The correct place to find the documentation for the YOLOv8-specific YAML configurations is in the YOLOv8 GitHub repository itself, specifically in the YOLOv8 model YAML file.

Feel free to let me know if you have any further questions.

**Comment by lesterlee89** [comment]
@glenn-jocher @Laughing-q ,if use the pretrained weights, the activation is invalid. is that a bug

**Comment by Laughing-q** [comment]
@lesterlee89 if you want to use a custom model yaml and meanwhile loading pretrained weights, you can create the model first and load the weights:
```python
from ultralytics import YOLO
model = YOLO("yolov8n-custom.yaml").load("yolov8n.pt")
```

**Comment by lesterlee89** [comment]
> @lesterlee89 if you want to use a custom model yaml and meanwhile loading pretrained weights, you can create the model first and load the weights:
> 
> ```python
> from ultralytics import YOLO
> model = YOLO("yolov8n-custom.yaml").load("yolov8n.pt")
> ```

thanks for you reply. i mean if use yolov8n.yaml without yolov8n.pt, and change activation to nn.ReLU() in the yolov8n.yaml, the activation is set ok. but if use yolov8n.yaml with yolov8n.pt as pretrained, whatever activation set, the model only use the nnSiLU() train. please let me know if i use something wrong

**Comment by lesterlee89** [comment]
@Laughing-q 
i find BottleneckCSP, ConvTranspose, RepConv also have default activation=nn.SiLU() , but in tasks.py it only handle Conv.

**Comment by Laughing-q** [comment]
@lesterlee89 yeah you're right, but official models don't use these models, we only use Conv and C2f modules which the latter is based on Conv so I think modifying activation would work with official architectures.

**Comment by Laughing-q** [comment]
@lesterlee89 I just tested it and it works properly I think.
here's my modified model yaml file(which I only modified the activation):
```yaml
# Ultralytics YOLO 🚀, AGPL-3.0 license
# YOLOv8 object detection model with P3-P5 outputs. For Usage examples see https://docs.ultralytics.com/tasks/detect

# Parameters
activation: nn.ReLU()
nc: 80  # number of classes
scales: # model compound scaling constants, i.e. 'model=yolov8n.yaml' will call yolov8.yaml with scale 'n'
  # [depth, width, max_channels]
  n: [0.33, 0.25, 1024]  # YOLOv8n summary: 225 layers,  3157200 parameters,  3157184 gradients,   8.9 GFLOPs
  s: [0.33, 0.50, 1024]  # YOLOv8s summary: 225 layers, 11166560 parameters, 11166544 gradients,  28.8 GFLOPs
  m: [0.67, 0.75, 768]   # YOLOv8m summary: 295 layers, 25902640 parameters, 25902624 gradients,  79.3 GFLOPs
  l: [1.00, 1.00, 512]   # YOLOv8l summary: 365 layers, 43691520 parameters, 43691504 gradients, 165.7 GFLOPs
  x: [1.00, 1.25, 512]   # YOLOv8x summary: 365 layers, 68229648 parameters, 68229632 gradients, 258.5 GFLOPs

# YOLOv8.0n backbone
backbone:
  # [from, repeats, module, args]
  - [-1, 1, Conv, [64, 3, 2]]  # 0-P1/2
  - [-1, 1, Conv, [128, 3, 2]]  # 1-P2/4
  - [-1, 3, C2f, [128, True]]
  - [-1, 1, Conv, [256, 3, 2]]  # 3-P3/8
  - [-1, 6, C2f, [256, True]]
  - [-1, 1, Conv, [512, 3, 2]]  # 5-P4/16
  - [-1, 6, C2f, [512, True]]
  - [-1, 1, Conv, [1024, 3, 2]]  # 7-P5/32
  - [-1, 3, C2f, [1024, True]]
  - [-1, 1, SPPF, [1024, 5]]  # 9

# YOLOv8.0n head
head:
  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 6], 1, Concat, [1]]  # cat backbone P4
  - [-1, 3, C2f, [512]]  # 12

  - [-1, 1, nn.Upsample, [None, 2, 'nearest']]
  - [[-1, 4], 1, Concat, [1]]  # cat backbone P3
  - [-1, 3, C2f, [256]]  # 15 (P3/8-small)

  - [-1, 1, Conv, [256, 3, 2]]
  - [[-1, 12], 1, Concat, [1]]  # cat head P4
  - [-1, 3, C2f, [512]]  # 18 (P4/16-medium)

  - [-1, 1, Conv, [512, 3, 2]]
  - [[-1, 9], 1, Concat, [1]]  # cat head P5
  - [-1, 3, C2f, [1024]]  # 21 (P5/32-large)

  - [[15, 18, 21], 1, Detect, [nc]]  # Detect(P3, P4, P5)
```
here's my python script:
```python
from ultralytics import YOLO
model = YOLO("./yolov8n-custom.yaml").load("yolov8n.pt")
model.export(format="onnx")
```
and here's the visualized onnx which looks the `ReLU` works:
![image](https://github.com/ultralytics/ultralytics/assets/61612323/38308f95-6319-42ee-a915-ec005607d0a0)

**Comment by lesterlee89** [comment]
@Laughing-q thank you. i use version 8.0.124, and use cli command “yolo detect train model=yolov8n.yaml pretrained=yolov8n.pt”, i will check if something wrong,

**Comment by glenn-jocher** [comment]
@lesterlee89 hello,

Thank you for raising the issue. Please note that the YOLOv8 repository does not provide code examples for modifying the activation function of the model in the YAML file. However, you can change the activation function easily by modifying the YAML file directly.

To change the activation function without using pretrained weights, you can simply modify the `activation` parameter in the YAML file to your desired activation function, such as `nn.ReLU()`.

If you are encountering issues when using pretrained weights, make sure you are using the correct version of the YOLOv8 repository (version 8.0.124) and check if any errors are occurring during the training process.

Please let us know if you need further assistance or if you have any additional questions.

Best regards.

**Comment by Sped0n** [comment]
For those who are curious or just want to know more details of this, I left my discovery here.

## TLDR

you **just have to load your custom model yaml while training**, other stuff are the same as the default process, you might not get what you want if you try exporting/inferrencing/validating the model with the loaded model yaml.

## Pkg Version

```
ultralytics        8.0.143
```

## Python scripts

### train

load your custom model yaml here

```python
from ultralytics import YOLO

if __name__ == "__main__":
    model = YOLO("/Users/spedon/eden/ml/yolov8/yolov8n.yaml").load(  # custom model yaml
        "/Users/spedon/eden/ml/yolov8/yolov8n.pt"  # pretrain weight from ultralytics
    ) 
    model.train(
        data="/Users/spedon/eden/ml/yolov8/datasets/TI2022/ti22.yaml",
        epochs=300,
        imgsz=416,
        device="mps",
        cache="disk",
        batch=64,
        optimizer="AdamW",
        save_period=5,
    )

```

### export

I try with loading custom model yaml here, but validation on exported onnx model get **no result**, instead, just use the weight file to export get expected mAP value in validation.

``` python
from ultralytics import YOLO

if __name__ == "__main__":
    model = YOLO(
        "/Users/spedon/eden/ml/yolov8/runs/detect/train2/weights/best.pt"
    ) 
    model.export(format="onnx", imgsz=416, opset=11, dynamic=False, simplify=True)
```

### inference

same here, no result if custom model yaml is loaded.

```python
from ultralytics import YOLO

if __name__ == "__main__":
    model = YOLO(
        "/Users/spedon/eden/ml/yolov8/runs/detect/train2/weights/best.pt"
    )
    results = model("./t.jpg", imgsz=416)
```

**Comment by glenn-jocher** [comment]
@Sped0n hello,

Thank you for sharing your discovery. It definitely provides valuable insights.

As you rightly pointed out, when training the model, load the custom model.yaml. The training process should then proceed as per usual.

We do advise to load the model only with the trained weights for the export and inference steps. As per your experience, loading the custom model.yaml during these stages could lead to unexpected results such as no output.

Always ensure to use the correct weight file path while performing export and inference.

We appreciate your contribution to the community and hope this information helps others!

Thank you.

**Comment by srishivathmika1809** [comment]
By changing the activation function to ReLU, will the softmax activation function will also change?

**Comment by glenn-jocher** [comment]
@srishivathmika1809 no, changing the activation function to ReLU wouldn't implicitly change the softmax activation function. These are two isolated activation functions applicable to different layers of the neural network. Softmax function is typically used as the activation function for the output layer in case of multi-class classification problems, while ReLU is more commonly used in hidden layers to introduce non-linearity and handle the vanishing gradient problem. If you want to change the softmax activation, you will need to do it explicitly.

**Comment by ovidedecroly** [comment]
@glenn-jocher Hello
what if we just want to replace the ReLU activation on one of the Convs in the yolov8 architecture?
Thanks

**Comment by glenn-jocher** [comment]
@ovidedecroly hello,

To replace the ReLU activation on one of the Convs in a specific part of the YOLOv8 architecture, such as just in the head and not in the backbone, you would need to modify the YAML architecture configuration file where the model definition is laid out.

In the YAML file, layers are defined block by block, and each Conv layer can have its own specified activation function. To change the activation for Conv layers only in the head section, you'd locate the Conv layers within the head part of the YAML file and change their 'act' parameter to `nn.ReLU()`, or whichever activation function you prefer.

Keep in mind that this configuration is sensitive; any changes should be done carefully to ensure the structure aligns with the YOLOv8 model's expectations. After modifying the YAML file, you can proceed to train or fine-tune your model with the updated configuration.

This flexibility allows you to experiment with different activations in different parts of the network to potentially improve performance on your specific dataset.

If you proceed with these changes, remember to verify the new architectural changes by training the model and observing whether the performance metrics meet your expectations.

Best regards.

**Comment by DevMSri** [comment]
What if I wanted to use a custom activation function defined in a python file, how would go about modifying the yolov8.yaml file such that this works?

**Comment by glenn-jocher** [comment]
@DevMSri yes, you can change the activation layer in the YOLOv8 model. YOLOv8, like other YOLO models, is defined by a configuration file in YAML format, which specifies the architecture of the model including the activation functions used in each layer.

To change the activation function, you would need to modify the `.yaml` file that defines the model architecture. Here's a general outline of the steps you would take:

1. Locate the `.yaml` file that defines your YOLOv8 model architecture. This file is typically found in the `models` directory of the Ultralytics YOLO repository.

2. Open the `.yaml` file with a text editor.

3. Find the layer(s) where you want to change the activation function. The activation functions are usually specified with a key called `activation`.

4. Replace the current activation function with the one you want to use. YOLOv8 supports several activation functions, such as `ReLU`, `LeakyReLU`, `SiLU` (also known as `Swish`), and `Mish`.

5. Save the `.yaml` file after making your changes.

6. Use the modified `.yaml` file to build and train your model.

Here's an example of what a layer definition might look like in the `.yaml` file before and after changing the activation function:

Before:
```yaml
- id: 3
  type: Conv
  filters: 256
  size: 3
  stride: 1
  activation: silu
```

After:
```yaml
- id: 3
  type: Conv
  filters: 256
  size: 3
  stride: 1
  activation: mish
```

In this example, the activation function for the convolutional layer with `id: 3` is changed from `silu` to `mish`.

Remember to train your model from scratch after changing the activation function to ensure that the weights are properly optimized for the new activations.

If you have any further questions or need assistance with a specific part of the process, feel free to ask!

**Comment by DevMSri** [comment]
Thank you for your reply but I have written a more specific post about my problem if you could please read that and then you might be able to understand my problem in a bit more detail. The link: https://github.com/ultralytics/ultralytics/issues/8243

**Comment by glenn-jocher** [comment]
@DevMSri to change the activation function in YOLOv8, you'll need to modify the model's architecture configuration file (`.yaml` file). The YOLOv8 architecture is defined in a YAML file where each layer's type and parameters are specified, including the activation function.

Here's a general approach to changing the activation function:

1. **Locate the Model Configuration File**: Find the `.yaml` file that defines the architecture of the YOLOv8 model you are using. This file will be in the format `yolov8*.yaml` (e.g., `yolov8n.yaml` for YOLOv8n).

2. **Modify the Activation Function**: Open the `.yaml` file in a text editor. Look for the `activation` parameter within the layer definitions. By default, YOLOv8 uses the SiLU (Sigmoid Linear Unit, also known as Swish) activation function. You can change it to another supported activation function by replacing `silu` with the desired activation function name.

   Here's an example of what a layer definition might look like before and after changing the activation function:

   ```yaml
   # Before
   - from: -1
     type: Conv
     args: [256, 3, 1, 1]
     activation: silu

   # After (changing to Leaky ReLU for example)
   - from: -1
     type: Conv
     args: [256, 3, 1, 1]
     activation: leaky_relu
   ```

3. **Supported Activation Functions**: YOLOv8 supports several activation functions, including `silu` (Swish), `relu`, `leaky_relu`, `mish`, and others. You can choose any of these supported functions as per your requirement.

4. **Save the Configuration File**: After making the changes, save the `.yaml` file.

5. **Train the Model**: Use the modified `.yaml` file to train your model on your custom dataset. The training process will now use the new activation function you specified.

6. **Evaluate the Performance**: After training, evaluate the performance of the model with the new activation function on your validation set to see if it has improved.

Remember that changing the activation function can significantly affect the model's performance and convergence behavior. It's a good idea to experiment with different functions and compare the results to determine the best choice for your specific application.

If you have any further questions or run into issues, feel free to reach out to the Ultralytics team or the community for assistance.

**Comment by DevMSri** [comment]
Thank you! That helps a lot but I wanted to use an activation function I defined in a python file in my yolov8.yaml file and I was wondering how I would go about that rather than using pre-built activation functions.

**Comment by glenn-jocher** [comment]
@DevMSri yes, you can change the activation layer in the YOLOv8 model. YOLOv8, like other YOLO models, is defined by a configuration file in YAML format, which specifies the architecture of the model including the activation functions used in each layer.

To change the activation function, you would need to modify the model's YAML configuration file. Here's a general approach to how you can do it:

1. Locate the YAML file that defines your YOLOv8 model architecture. This file typically has a structure that defines various layers and their parameters.

2. In the YAML file, find the layer where you want to change the activation function. The activation functions are usually specified with a key such as `activation`.

3. Replace the current activation function with the one you want to use. Make sure that the activation function you want to use is supported by the Ultralytics framework.

4. Save the YAML file after making the changes.

5. Load the modified YAML file when creating a new YOLO model instance for training.

Here's an example of what a part of the YAML configuration file might look like:

```yaml
# Example layer definition before modification
- type: Conv
  filters: 256
  size: 3
  stride: 1
  activation: leaky  # This is the activation function you want to change
```

To change the activation function to, say, `ReLU`, you would modify it like this:

```yaml
# Example layer definition after modification
- type: Conv
  filters: 256
  size: 3
  stride: 1
  activation: relu  # Changed to ReLU activation
```

The available activation functions in the Ultralytics YOLO framework typically include `ReLU`, `LeakyReLU`, `Mish`, `SiLU` (Swish), etc. You can check the Ultralytics codebase or documentation for a complete list of supported activation functions.

Remember to test the model after changing the activation function to ensure that it is working as expected. Changing activation functions can have a significant impact on model performance and convergence during training.
